ext/dcos-installer/dcos_installer/action_lib/__init__.py

```diff
@@ -151,10 +164,10 @@ def _remove_host(state_file, host):
 
 
 @asyncio.coroutine
-def install_dcos(config, block=False, state_json_dir=None, hosts=[], async_delegate=None, try_remove_stale_dcos=False,
-                 roles=None, **kwargs):
-    if roles is None:
-        roles = ['master', 'agent']
+def install_dcos(config, block=False, state_json_dir=None, hosts=None, async_delegate=None, try_remove_stale_dcos=False,
+                 **kwargs):
+    if hosts is None:
+        hosts = []
```

Comment:
basically it is a bad practice to have mutable default parameter, this is a workaround to make sure if `hosts` wasn't passed, we should always default to a new list.